### ISSUE TITLE
[cancel_invoice] send_payment will cause node2 return Musig2VerifyError when invoice cancel 

### ISSUE THREAD

**Issue by gpBlockchain** [nervosnetwork/fiber#305]
1. node2 new_invoice
2. node2 cancel_invoice
3. node1 send_payment
node2 return Musig2VerifyError
```
  2024-11-13T14:39:12.254834Z ERROR fnn::fiber::channel: Error while processing channel message: Musig2VerifyError(BadSignature)

```
[node2.log](https://github.com/user-attachments/files/17734223/node.log)
[node.log](https://github.com/user-attachments/files/17734273/node.log)

**Comment by gpBlockchain** [comment]
The same thing will happen when the invoice time approaches expiry

**Comment by chenyukang** [comment]
> The same thing will happen when the invoice time approaches expiry

which branch are you running, the expiry check logic haven't been merged, it's on https://github.com/chenyukang/fiber/pull/2

**Comment by gpBlockchain** [comment]
> > The same thing will happen when the invoice time approaches expiry
> 
> which branch are you running, the expiry check logic haven't been merged, it's on [chenyukang#2](https://github.com/chenyukang/fiber/pull/2)

main branch

**Comment by chenyukang** [comment]
https://github.com/chenyukang/fiber/tree/debugging-sig-issue
this branch can reproduce the issue, it's the different cause, but I believe it's the same kind of issue for `Musig2VerifyError`.

run command : `npm exec -- @usebruno/cli@1.20.0 run e2e/router-pay -r --env test`

check the logs.
